$$\require{mhchem}$$

# Adiabatic Flame Temperature: Constant $C_p$

The adiabatic flame temperature is the temperature achieved when a given fuel-oxidizer combination is combusted in a perfectly insulated reactor. If no energy is lost to the surroundings via heat transfer and the reaction occurs at constant pressure, conservation of energy dictates that the total enthalpy of the reactants is equal to the total enthalpy of the products. Defining _state 1_ as the reactants and _state 2_ as the products, we can write:

$$h_{\text{mix}, 2} = h_{\text{mix}, 1}$$

The enthalpy of the mixture has two components:

1. the enthalpy of formation: $\Delta^{\circ}_{\text{f}}h$ 
2. the sensible enthalpy: $h_i$

To make this more concrete, let's pick a specific reaction: stoichiometric combustion of methane, $\ce{CH4}$ with air:

$$\ce{CH4 + 2 O2 + $7.52$N2 <=> 2 H2O + CO2 + $7.52$N2}$$

From the reaction stoichiometry (note that the net stoichiometric coefficient for $\ce{N2}$ is zero), the heat of combustion released by this reaction is:

$$\begin{aligned}q_{\text{rxn}} &= -\sum_k \nu_k \Delta^{\circ}_{\text{f}}h_k\\
&= \Delta^{\circ}_{\text{f}}h_{\rm CH_4} + 2\Delta^\circ_{\rm f}h_{\rm O_2} - \Delta^\circ_{\rm f}h_{\rm CO_2}-2 \Delta^\circ_{\rm f}h_{\rm H_2O}\end{aligned}$$

where $\nu_k$ is the stoichiometric coefficient of species $k$, which is positive for products and negative for reactants. By definition, the heat of reaction is released at the reference temperature.

However, since the reaction must occur at constant enthalpy, the heat released must be stored in the products. In essence, the heat is used to heat the products from the initial, reference, temperature to a final temperature:

$$\begin{aligned}q_{\text{heating}} &= \sum_k\nu_{\text{prod},k}\left[h_k\left(T_2\right) - h_k\left(T_1\right)\right]\\
&= h_{\ce{CO2}}\left(T_2\right) - h_{\ce{CO_2}}\left(T_1\right) + 2\left[h_{\ce{H2O}}\left(T_2\right) - h_{\ce{H2O}}\left(T_1\right)\right] + 7.52\left[h_{\ce{N2}}\left(T_2\right) - h_{\ce{N_2}}\left(T_1\right)\right]\end{aligned}$$

where, again, states 1 and 2 represent the final and initial states, respectively. $\nu_{\text{prod}}$ is the product stoichiometric coefficient. Note that this assumes complete combution - there are no reactant species left to heat.

For a purely adiabatic reaction, we will have $q_{\rm rxn} = q_{\rm heating}$, and hence:

$$\sum_k\nu_{\text{prod},k}\left[h_k\left(T_2\right) - h_k\left(T_1\right)\right] + \sum_k \nu_k \Delta^{\circ}_{\text{f}}h_k = 0$$

or:

$$h_{\ce{CO2}}\left(T_2\right) - h_{\ce{CO_2}}\left(T_1\right) + 2\left[h_{\ce{H2O}}\left(T_2\right) - h_{\ce{H2O}}\left(T_1\right)\right] + 7.52\left[h_{\ce{N2}}\left(T_2\right) - h_{\ce{N_2}}\left(T_1\right)\right] + \Delta^{\circ}_{\text{f}}h_{\rm CH_4} + 2\Delta^\circ_{\rm f}h_{\rm O_2} - \Delta^\circ_{\rm f}h_{\rm CO_2}-2 \Delta^\circ_{\rm f}h_{\rm H_2O} = 0$$

Note that only a few variables are dependent on the final temperature $T_2$.  We might be tempted to write the above equation in terms of those variables, but let's hold off for now. Our next step really depends on what method we use to solve the problem. In short, we need a means of calculating the enthalpy values, so that we may determine which value of $T_2$ satisfies our adiabatic condition.  

## JANAF Tables

One popular source for the enthalpy values are the [JANAF tables](https://janaf.nist.gov/), a thermodynamic database maintained by NIST. These provide a range of thermodynamic properties, including enthalpies of formation, total sensible enthalpy, and others, for a range of temperatures and for a wide variety of species. For the curious, a collection of these tables are stored in the `data` folder that accompanies this notebook.

We will take the enthalpies of formation at the initial condition of 298.15 K:

|   Species  | $\Delta^{\circ}_{\text{f}}h_k\left(298.15\,\mathrm{K}\right))\,\frac{\mathrm{kJ}}{\mathrm{mol}}$ |
|------------|--------------------------------------------------------------------------------------------------|
| $\ce{CH4}$ | -74.873                                                                                          |
| $\ce{O2}$  | 0.0                                                                                              |
| $\ce{N2}$  | 0.0                                                                                              |
| $\ce{CO2}$ | -393.522                                                                                         |
| $\ce{H2O}$ | -241.826                                                                                         |

From these, let's solve for the heat released by the reaction.  Give it the variable name `q_rxn`:

In [3]:
# Entahlpies of formation, kJ/mol
h_f_CH4 = -74.873
h_f_CO2 = -393.522
h_f_H2O = -241.826
h_f_O2 = 0

q_rxn = h_f_CO2 + 2 * h_f_H2O - h_f_CH4 - 2*h_f_O2

print(f"The heat of reaction is {q_rxn:.2f} kJ/mol")

This heat must be stored in the sensible enthalpy of the products, which reach an elevated temperature $T_2$, after combustion.

To calculate the final temeprature, we need to estimate how the species enthalpies change as a function of temperature. The JANAF tables actually give us $h_k(T) - h_k(298.15\,\mathrm{K})$ as a function of $T$, allowing for a high-precision solution. This is demonstrated in another [Notebook using interpolation](./ad-flame-T-interpolated.ipynb).

However, a simpler and nearly as accurate solution uses the definition of the specific heat at constant pressure to calculate the change in enthalpy. $C_p$ is defined such that:

$$h_k(T_2) - h_k(T_1) = \int_{T_1}^{T_2}C_{p,k} dT$$

If we assume $C_{p,k}$ is constant, this reduces to:

$$h_k(T_2) - h_k(T_1) = C_{p,k}\left(T_2 - T_1\right)$$

While $C_p$ is not truly constant, we can take the value at some intermediate temperature and assume that the variation in $C_p$ from this average value is not _too_ significant. This simplifies the calculation because we can use average values of $C_p$ for each species.

Finally, to simplify a little bit further we note that nitrogen makes up the bulk of the products on a molar basis. For instance, the mole fraction of nitrogen in the products is:

$$X_{\ce{N2}} = \frac{7.52}{10.52} = 0.715$$

Therefore, we will assume that the change in sensible enthalpy of all of the products can be approximated by using an average $C_p$ value of nitrogen:

$$q_{\text{heating}} = \sum_k\nu_{\text{prod},k}\left[h_k\left(T_2\right) - h_k\left(T_1\right)\right] \approx C_{p,\ce{N2}} n_{\text{tot}} \left(T_2 - T_1\right)$$

where $n_{\text{tot}} = 10.52$ for the products.

So let's consult the JANAF tables once more to find a value for $C_p$. We guess that the final temperature will be around 2000 K, so an intermediate temperature of $T_{\text{avg}} =$ 1000 K is sufficiently accurate here.

$$C_{p,\ce{N2}} = 32.697\,\frac{\text{J}}{\text{mol K}}$$

Now we can solve algebraically for $T_2$:

$$T_2 = T_1 - \frac{q_{\text{rxn}}}{n_{\text{tot}}C_{p,\text{avg}}}$$

In [7]:
T_1 = 298.15  # Initial temperature, K
C_p_avg = 32.697E-3  # Convert to kJ/mol-K
n_tot = 10.52

T_2 = T_1 - q_rxn / n_tot / C_p_avg
T_avg = (T_1 + T_2) / 2
print(f"Initial T_ad = {T_2:.2f} K")
print(f"T_avg = {T_avg:.2f} K")

Initial T_ad = 2630.61 K
T_avg = 1464.38 K


Not bad! If this is correct, the average temperature is 1464.4 K. So the average $C_p$ at 1000 K is perhaps a bit too low. Let's try to improve the accuracy by using:

$$C_{p,\ce{N2}}(1400\,\mathrm{ K})=34.518\,\frac{\mathrm{J}}{\text{mol K}}$$

Right away, we can see that $C_p$ hasn't changed all _that_ much, in going from 1000 K to 1400 K.  But let's check, anyway:

In [9]:
C_p_avg = 34.518E-3  # Convert to kJ/mol-K

T_2 = T_1 - q_rxn/n_tot/C_p_avg
T_avg = (T_1 + T_2) / 2
print(f"Improved T_ad = {T_2:.2f} K")
print(f"T_avg = {T_avg:.2f} K")

Improved T_ad = 2507.56 K
T_avg = 1402.85


We see that the adiabatic flame temperature estimate dropped by 123 K, and now our actual average temperature is quite close to our guess value.

The detailed approach mentioned previously ([ad-flame-T-interpolated](./ad-flame-T-interpolated.ipynb)) returns an adiabatic flame temperature of $T_{\text{ad}} =$ 2327 K, which is not _too_ far off from our rather simple estimate here!